sklearn_cotraining/classifiers.py

```diff
@@ -128,11 +127,7 @@ def fit(self, X1, X2, y):
 			L.extend([U_[x] for x in p])
 			L.extend([U_[x] for x in n])
 
-			#TODO: optimize these removals from U_
-			#this is currently (2p + 2n)O(n)
-			#and I think it can be reduced to O(n) rather easily
-			for i in p: U_.pop(i)
-			for i in n: U_.pop(i)
+			U_ = [elem for elem in U_ if (elem in p or elem in n)]
```

Comment:
Should we be populating U_ with new samples (i.e. samples that are not in p or n - which were just added into L)?

Reply:
You are right. That line was supposed to remove all elements of p and n from U_. That has been fixed.